local-pv-provisioner/controllers/persistent_volume.go

```diff
@@ -0,0 +1,58 @@
+package controllers
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/hex"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+const (
+	storageClass     = "local-storage"
+	hostNameLabelKey = "kubernetes.io/hostname"
+)
+
+func (dd *DeviceDetector) pvName(devName string) string {
+	hasher := sha1.New()
+	hasher.Write([]byte(devName))
```

Comment:
Use `sha1.Sum` for better performance.
https://golang.org/pkg/crypto/sha1/#Sum

But why don't you use `devName` as is?

Reply:
Removed sha1 because devName's rule was changed.